### PR TITLE
Add support for external Dynflow core 

### PR DESCRIPTION
**Pull request by adamruzicka**
this option is being introduced in https://github.com/theforeman/smart_proxy_dynflow/pull/63

### CODE REVIEW

File: manifests/plugin/dynflow.pp [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #
 # $open_file_limit::       Limit number of open files - Only Red Hat Operating Systems with Software Collections.
 #
+# $external_core::         Forces usagge of external/internal Dynflow core
```
Comment (ekohl) [inline manifests/plugin/dynflow.pp]:
```suggestion
# $external_core::         Forces usage of external/internal Dynflow core
```

**Comment by adamruzicka** [comment]
I'm currently looking it. I most likely won't push it into this PR, but having all the platforms unified somewhere down the road would be nice.

**Comment by iNecas** [comment]
In that case I guess we need to have the external_core set only for el-based systems.

**Comment by adamruzicka** [comment]
That makes sense. is that a thing which needs to be done here or should it go to installer/scenarios/somewhere?

**Comment by iNecas** [comment]
If possible to do this changes in a way that they would work with the old installer and it would not be too much of a hassle, I'm for doing the changes changes here

File: templates/plugin/dynflow.yml.erb [hunk 2]
```diff
@@ -2,3 +2,13 @@
 :enabled: <%= @module_enabled %>
 :database: <%= scope.lookupvar('::foreman_proxy::plugin::dynflow::database_path') %>
 :core_url: <%= scope.lookupvar('::foreman_proxy::plugin::dynflow::core_url') %>
+
+# If true, external core will be used even if the core gem is available
+# If false, the feature will be disabled if the core gem is not available
+# If unset, the process will fallback to external core if the core gem is not available
```
Comment (ekohl) [inline templates/plugin/dynflow.yml.erb]:
```suggestion
# If unset, the process will fallback to autodetection, using external core if the core gem is unavailable
```

File: templates/plugin/dynflow.yml.erb [hunk 3]
```diff
@@ -2,3 +2,13 @@
 :enabled: <%= @module_enabled %>
 :database: <%= scope.lookupvar('::foreman_proxy::plugin::dynflow::database_path') %>
 :core_url: <%= scope.lookupvar('::foreman_proxy::plugin::dynflow::core_url') %>
+
+# If true, external core will be used even if the core gem is available
+# If false, the feature will be disabled if the core gem is not available
```
Comment (ekohl) [inline templates/plugin/dynflow.yml.erb]:
```suggestion
# If false, the feature will be disabled if the core gem is unavailable
```

**Pull request merged**